File: mail-go/internal/services/mail_service.go
package services

import (
	"fmt"
	"log"
	"net/smtp"
	"os"
)

func SendEmail(course, email, company, phone, participantsDesired, date string) error {
	username := os.Getenv("SMTP_USERNAME")
	password := os.Getenv("SMTP_PASSWORD")
	smtpHost := os.Getenv("SMTP_HOST")
	smtpPort := os.Getenv("SMTP_PORT")
	from := "mailtrap@demomailtrap.com"
	to := "koffep@gmail.com"

	smtpUrl := fmt.Sprintf("%s:%s", smtpHost, smtpPort)

	auth := smtp.PlainAuth("", username, password, smtpHost)

	message := fmt.Sprintf(
		"To: %s\r\n"+
			"From: %s\r\n"+
			"Subject: Course Information\r\n"+
			"Content-Type: text/html; charset=\"UTF-8\"\r\n"+
			"\r\n"+
			"<html><body>"+
			"<p>Course: %s<br>"+
			"Email: %s<br>"+
			"Company: %s<br>"+
			"Phone: %s<br>"+
			"Participants Desired: %s<br>"+
			"Date: %s<br>"+
			"</p></body></html>",
		to, from, course, email, company, phone, participantsDesired, date)

	err := smtp.SendMail(smtpUrl, auth, from, []string{to}, []byte(message))
	if err != nil {
		log.Println("Failed to send email:", err)
		return err
	}

	log.Println("Email sent successfully")
	return nil
}

File: mail-rust/mail/src/services/mail_service.rs
use std::env;
use std::io::Error;

use lettre::{Message, SmtpTransport, Transport};
use lettre::transport::smtp::authentication::Credentials;

pub async fn send_email(course: &String, email: &String, company: &String, phone: &String, participants_desired: &String, date: &String) -> Result<(), Error> {
    let username = env::var("SMTP_USERNAME").expect("env not set");
    let password = env::var("SMTP_PASSWORD").expect("env not set");
    let smtp_host = env::var("SMTP_HOST").expect("env not set");
    let smtp_port = env::var("SMTP_PORT").expect("env not set");
    let from = "mailtrap@demomailtrap.com";
    let to = "koffep@gmail.com";

    let email = Message::builder()
        .from(from.parse().unwrap())
        .to(to.parse().unwrap())
        .subject("Course Information")
        .header(lettre::message::header::ContentType::TEXT_HTML)
        .body(format!(
            "<html><body>
            <p>Course: {}<br>
            Email: {}<br>
            Company: {}<br>
            Phone: {}<br>
            Participants Desired: {}<br>
            Date: {}<br>
            </p></body></html>",
            course, email, company, phone, participants_desired, date))
        .unwrap();

    let creds = Credentials::new(username, password);

    let mailer = SmtpTransport::starttls_relay(&*smtp_host)
        .unwrap()
        .port(smtp_port.parse().unwrap())
        .credentials(creds)
        .build();

    match mailer.send(&email) {
        Ok(_) => println!("Email sent successfully!"),
        Err(e) => eprintln!("Could not send email: {:?}", e),
    }

    Ok(())
}